### PR TITLE
Benchmark and strengthen Turso cache wrapper hot paths

### LINKED ISSUE #595: perf(test,bench): benchmark and strengthen Turso cache wrapper hot paths
### Summary
Expand correctness coverage and add benchmarks for `memory-storage-turso/src/cache/wrapper.rs`, which appears to be a likely performance-sensitive module.

### Problem
`memory-storage-turso/src/cache/wrapper.rs` currently has only 5.65% HEAD coverage, although the patch coverage for touched lines is 100.00%.

### Why this matters
This suggests the immediate patch was exercised, but the broader module remains weakly validated. Because the cache wrapper sits between higher-level retrieval logic and storage access, it is a likely hotspot for latency, allocation churn, or behavioral edge cases.

### Scope
- Add correctness tests for the cache wrapper.
- Add Criterion benchmarks for common access patterns.
- Record whether the observed cost is primarily cache logic, serialization, or storage I/O.

### Proposed work
- Add tests for cache miss followed by fill.
- Add tests for warm-cache repeated reads.
- Add tests for empty or absent values.
- Add tests for invalidation or refresh behavior if supported.
- Add tests for propagation of underlying storage failures.
- Add Criterion benchmarks for cold reads, warm reads, and repeated mixed-key access.
- Optionally compare small and medium payload sizes if serialization overhead looks material.

### Acceptance criteria
- `wrapper.rs` gains meaningful correctness coverage beyond the currently touched lines.
- A benchmark suite exists for core cache access patterns.
- Findings are documented in the issue or linked plan note.
- A follow-up optimization issue can reference measured results instead of assumptions.

### Notes
Do not wire full benchmark execution into every PR initially. Start with local or dedicated workflow execution, then add a lightweight regression check only if the signal is useful.

### PR DESCRIPTION
**Pull request by d-o-hub**
I have expanded the test coverage and benchmarking suite for the Turso cache wrapper. 

Key changes:
1.  **Correctness Coverage**: Added `memory-storage-turso/src/cache/tests/wrapper_edge_cases.rs` which specifically tests edge cases like cache fill after miss, handling of absent values (ensuring they aren't incorrectly cached), and propagation of underlying storage errors.
2.  **Benchmark Strengthening**: Updated `benches/cache_benchmarks.rs` to include:
    - `bench_episode_cold_read`: Measures the latency of the first access (miss + fill).
    - `bench_mixed_access_80_20`: Simulates a more realistic workload with mixed hits and misses.
    - `bench_payload_size_comparison`: Compares the cache overhead for small vs. large (100 steps) episodes.
3.  **Refactoring**: Moved internal tests from `wrapper.rs` to `wrapper/tests.rs` to align with the crate's testing patterns and improve maintainability.

These changes provide better visibility into the performance characteristics of the cache wrapper and ensure its behavior is well-validated across various scenarios.

Fixes #595

---
*PR created automatically by Jules for task [4253425837838612657](https://jules.google.com/task/4253425837838612657) started by @d-o-hub*

### CODE REVIEW

File: benches/cache_benchmarks.rs [hunk 1]
```diff
@@ -155,6 +173,134 @@ fn bench_episode_retrieval_cached_10(c: &mut Criterion) {
...
+            let cached = CachedTursoStorage::new(storage, CacheConfig::default());
+
+            let episode_id = Uuid::new_v4();
+            let episode = create_test_episode(episode_id);
+
+            rt().block_on(async {
+                cached.store_episode_cached(&episode).await.unwrap();
+
+                // Benchmark first access - cache miss + fill
+                let start = std::time::Instant::now();
```
[the remark this answers is not in the document]
Comment (d-o-hub) [reply benches/cache_benchmarks.rs]: @jules

File: benches/cache_benchmarks.rs [hunk 2]
```diff
@@ -155,6 +173,134 @@ fn bench_episode_retrieval_cached_10(c: &mut Criterion) {
...
+                black_box(start.elapsed());
+            });
+        });
+    });
+
+    group.finish();
+}
+
+/// Compare cache overhead for small vs large episodes
+fn bench_payload_size_comparison(c: &mut Criterion) {
```
[the remark this answers is not in the document]
Comment (d-o-hub) [reply benches/cache_benchmarks.rs:253]: @jules
